packages/documentsearch-extension/src/index.ts

```diff
@@ -63,6 +64,13 @@ const labShellWidgetListener: JupyterFrontEndPlugin<void> = {
       }
       if (registry.hasProvider(widget)) {
         widget.addClass(SEARCHABLE_CLASS);
+        // Only bind Esc in case the document is currently searchable,
+        // else it will clash with Esc to close completer on fileeditor
+        app.commands.addKeyBinding({
+          command: CommandIDs.end,
+          keys: ['Escape'],
+          selector: '.jp-mod-searchable'
+        });
```

Comment:
I don't believe it is doing what you think it does. Once the command is added it will remain there forever. Hence this will only "solve" the problem for the very dismissals of completer which happen before first opening of search panel. Otherwise, it is the same as defining it in schema, but prevents user customisation, so we should not do it.

Reply:
I guess one could remove the command each time the search widget is closed, each time the texteditor widget loses the "searchable" class... But I guess it's not coherent with the overall style. 
I'm guessing you can use or not use whatever commit of mine that you find useful, right, or do you want me to do a rollback to the version with the dismiss command for the completer?